src/main/java/com/dataloom/neuron/Neuron.java

```diff
@@ -51,14 +78,65 @@ public void activateReceptor( SignalType type, Receptor receptor ) {
         }
     }
 
-    public void transmit( AuditableSignal signal ) {
+    public void transmit( Signal signal ) {
+
+        // 1. write to the Audit Entity Set
+        UUID auditId = writeToAuditEntitySet( signal );
+
+        if ( auditId != null ) {
 
-        // 1. audit event
-        this.auditLogQueryService.store( signal );
+            // 2. write to the Audit Log Table
+            writeToAuditLog( signal, auditId );
+        }
+
+        // 3. hand off event to receptors
+        Set<Receptor> receptors = this.receptors.get( signal.getType() );
+
+        // TODO: does order matter?
+        // TODO: what about parallelization?
+        receptors.forEach( receptor -> receptor.process( signal ) );
+    }
 
-        // 2. hand off event to receptors
-        // List<Receptor> receptors = this.receptors.get( signal.getType() );
-        // receptors.forEach( synapse -> synapse.process( signal ) );
+    private UUID writeToAuditEntitySet( Signal signal ) {
+
+        try {
+
+            UUID auditEntitySetId = AuditEntitySet.getId();
+            UUID auditEntitySetSyncId = dataSourceManager.getCurrentSyncId( auditEntitySetId );
```

Comment:
Got it. I agree, it's a good idea to store the `syncId` be inside `AuditEntitySet` and avoid unnecessarily overloading Hazelcast.

Your suggestion is good, but I think it's not enough. I'm not sure what the order of operations is, but isn't it theoretically possible that you'll invoke `AuditEtitySet.setSyncId()` before the EntitySet is actually initialized, therefore setting `syncId` to something not related to the Audit EntitySet?

It seems like we'll need the Audit EntitySet name as well...?